source/dicom/valuerep.py

```diff
@@ -48,6 +48,20 @@ class DSfloat(float):
     """
     __slots__ = 'original_string'
 
+    def __getstate__(self):
+        if type(self.__slots__) is str and hasattr(self, self.__slots__):
```

Comment:
You probably should be checking this as `isinstance(self.__slots__, basestring)`.